src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -0,0 +1,663 @@
+                   |
+                   |Fastqs and read structures for read one, read two, i7 read, and i5 read should be given.  The read structures
+                   |may contain sample barcode bases ('B'), molecular identifier bases ('M'), template bases ('T'), and bases to
+                   |skip ('S'). Template bases should only be found on read one or read two.  Any molecular identifiers will be
+                   |concatenated using the '-' delimiter and placed in the given SAM record tag ("RX" by default).  The order of
+                   |concatenation will be read one, read two, i7 read, and i5 read, only considering reads with molecular identifiers.
+                   |Similarly, the sample barcode bases from the given read will be placed in the "BC" tag, using the same rules as
+                   |molecular identifiers, but applied to sample barcodes.
+                   |
+                   |The sample barcode for each sample in the sample sheet will be compared against the sample barcode bases in
+                   |each read, to assign each read to a sample.  Reads that to not match any sample within the given error tolerance
+                   |will be placed in the 'unmatched' file.
+                   |
+                   |The output directory will contain one BAM file per sample in the sample sheet, plus a BAM for reads that could
+                   |not be assigned to a sample given the criteria.  The output files will be the concatenation of sample id, sample
+                   |name, and sample barcode bases (expected not observed), delimited by ".".  A metrics file will also be output
+                   |for sample barcodes.  More information about these metrics can be found here:
+                   |  https://broadinstitute.github.io/picard/picard-metric-definitions.html#SampleBarcodeMetric
+                   |
+                   |The read group's sample id, sample name, and library id all correspond to the similarly named values in the sample sheet.
+                   |Library id will be the sample id if not found, and the platform unit will be the sample name concatenated with the sample
+                   |barcode bases delimited by a ".".
+                   |
+                   |The sample section of the sample sheet should contain information related to each sample with the following keys and values:
+                   |  - Sample Identifier:  Sample_ID
+                   |  - Sample Name:        Sample_Name
+                   |  - Library Identifier: Library_ID
+                   |  - Sample Project:     Sample_Project
+                   |  - Description:        Description
+                   |
+                   |The following are optional values may included in the sample sheet to specify sample barcode bases in any read:
+                   |  - Read One Inline Sample Barcode Bases*: R1_Barcode_Bases
```

Comment:
Should we use `Index` to match the standard fields?  E.g. `R1Index` and `R2Index`?